api/controllers/autocomplete.go

```diff
@@ -10,189 +10,154 @@ import (
 								{Key: "academic_session", Value: "$_id.academic_session"},
-							},
-						},
-						{Key: "sections",
-							Value: bson.D{
-								{Key: "$push",
-									Value: bson.D{
-										{Key: "section_number", Value: "$_id.section_number"},
-										{Key: "professors", Value: "$professor"},
-									},
-								},
+								{Key: "sections", Value: "$sections"},
 							},
 						},
 					},
 				},
 			},
-
-			bson.D{
-				{Key: "$group",
+		}},
+		{{Key: "$group",
+			Value: bson.D{
+				{Key: "_id",
 					Value: bson.D{
-						{Key: "_id",
-							Value: bson.D{
-								{Key: "subject_prefix", Value: "$_id.subject_prefix"},
-								{Key: "course_number", Value: "$_id.course_number"},
-							},
-						},
-						{Key: "academic_sessions",
-							Value: bson.D{
-								{Key: "$push",
-									Value: bson.D{
-										{Key: "academic_session", Value: "$_id.academic_session"},
-										{Key: "sections", Value: "$sections"},
-									},
-								},
-							},
-						},
+						{Key: "subject_prefix", Value: "$_id.subject_prefix"},
 					},
 				},
-			},
-
-			bson.D{
-				{Key: "$group",
+				{Key: "course_numbers",
 					Value: bson.D{
-						{Key: "_id",
+						{Key: "$push",
 							Value: bson.D{
-								{Key: "subject_prefix", Value: "$_id.subject_prefix"},
-							},
-						},
-						{Key: "course_numbers",
-							Value: bson.D{
-								{Key: "$push",
-									Value: bson.D{
-										{Key: "course_number", Value: "$_id.course_number"},
-										{Key: "academic_sessions", Value: "$academic_sessions"},
-									},
-								},
+								{Key: "course_number", Value: "$_id.course_number"},
+								{Key: "academic_sessions", Value: "$academic_sessions"},
 							},
 						},
 					},
 				},
 			},
-
-			bson.D{
-				{Key: "$project",
-					Value: bson.D{
-						primitive.E{Key: "_id", Value: 0},
-						{Key: "subject_prefix", Value: "$_id.subject_prefix"},
-						{Key: "course_numbers", Value: "$course_numbers"},
-					},
-				},
+		}},
+		{{Key: "$project",
+			Value: bson.D{
+				{Key: "_id", Value: 0},
+				{Key: "subject_prefix", Value: "$_id.subject_prefix"},
+				{Key: "course_numbers", Value: "$course_numbers"},
 			},
-		}
-
-		// get cursor for aggregation pipeline query results
-		// pipeline aggregated against the courses collection
-		cursor, err := courseCollection.Aggregate(ctx, autocompletePipeline)
-		if err != nil {
-			c.JSON(http.StatusInternalServerError, responses.AutocompleteResponse{Status: http.StatusInternalServerError, Message: "error", Data: err.Error()})
-			return
-		}
+		}},
+	}
 
-		// retrieve and parse all valid documents
-		if err = cursor.All(ctx, &autocompleteDAG); err != nil {
-			panic(err)
-		}
+	cursor, err := courseCollection.Aggregate(ctx, autocompletePipeline)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, responses.AutocompleteResponse{Status: http.StatusInternalServerError, Message: "error", Data: err.Error()})
+		return
+	}
 
-		// return result
```

Comment:
Getting rid of explicit type names (i.e. `primitive.E`) and comments doesn't make code cleaner, it makes its functionality less clear. Try to avoid doing it.